docs/upgrading-to-v0.18.x.md

```diff
@@ -0,0 +1,209 @@
+# Upgrading to v0.18.x
+
+Starting with `v0.18.0` version, [provider-gcp] will use `v1` version of GCP
+APIs for container group instead of `v1beta1`. Users relying on beta features, 
+suggested to use [provider-gcp-beta] instead, which could be installed
+side-by-side with [provider-gcp]. For more details on this, please see
+[RFC issue for provider-gcp-beta] and description of [the PR switching v1].
```

Comment:
It might be worth making it extra clear that this is different from `v1` and `v1beta1` Kubernetes APIs. It already says 'GCP APIs' but I initially read this as saying we were bumping our managed resource APIs to `v1`.

Reply:
Done.